crates/wasm-mutate/README.md

```diff
@@ -0,0 +1,141 @@
+<div align="center">
+  <h1><code>wasm-mutate</code></h1>
+
+<strong>A <a href="https://bytecodealliance.org/">Bytecode Alliance</a> project</strong>
+
+  <p>
+    <strong>wasm-mutate is a new tool for fuzzing Wasm compilers, runtimes, validators, and other Wasm-consuming programs.</strong>
+  </p>
+
+  <!-- TODO add proper links --> 
+  <p>
+    <a href="https://crates.io/crates/wasm-mutate"><img src="https://img.shields.io/crates/v/wasm-mutate.svg?style=flat-square" alt="Crates.io version" /></a>
+    <a href="https://crates.io/crates/wasm-mutate"><img src="https://img.shields.io/crates/d/wasm-mutate.svg?style=flat-square" alt="Download" /></a>
+    <a href="https://docs.rs/wasm-mutate/"><img src="https://img.shields.io/static/v1?label=docs&message=wasm-mutate&color=blue&style=flat-square" alt="docs.rs docs" /></a>
+  </p>
+</div>
+
+<!-- .  -->
+
+
+## Usage
+
+Add this to your `Cargo.toml`:
+
+```toml
+[dependencies]
+wasm-mutate = "0.1.0"
+```
+
+You can mutate a WebAssembly binary by using the cli tool:
+
+```bash
+./wasm-mutate original.wasm --seed 0 -o out.wasm --preserve-semantics
+```
+
+## Features
+
+* **semantically equivalent transformations:** `wasm-mutate` has the ability to
+  only apply semantics-preserving changes to the input Wasm module. When it is
+  used in this mode, the mutated Wasm computes identical results when
+  given the same inputs as the original Wasm module.
+* **determinism:** `wasm-mutate` is deterministic, i.e., given the same input
+  Wasm module and the same seed, it always produces the same mutated
+  output Wasm module.
+* **libfuzzer integration**: `wasm-mutate` integrates well with mutation-based fuzzers like libFuzzer. It
+  reuses the fuzzer's raw input strings. `wasm-mutate` works with the
+  `LLVMFuzzerCustomMutator` hook and the
+  `libfuzzer_sys::fuzz_mutator!` macro.
+  ### Example
+  ```rust
+  #![no_main]
+
+    use libfuzzer_sys::fuzz_target;
+    use wasmparser::WasmFeatures;
+
+    fuzz_target!(|bytes: &[u8]| {
```

Comment:
This example isn't using the `fuzz_mutator!` macro like the bullet point describes.